### PR TITLE
Add support for STM32F469XI and STM32F469I Discovery Kit

### PR DESCRIPTION
**Pull request by superna9999**
This PR fixes support for the STM32F469XI SoC and support for the STM32F469I Discovery kit board.

Documentation for the STM32F469I Discovery kit board is added based on the STM32F4DISCOVERY format.

### CODE REVIEW

File: boards/arm/stm32f469i_disco/stm32f469i_disco_defconfig [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# produce 180MHz clock at PLL output
+CONFIG_CLOCK_STM32_PLL_M_DIVISOR=8
+CONFIG_CLOCK_STM32_PLL_N_MULTIPLIER=360
+CONFIG_CLOCK_STM32_PLL_P_DIVISOR=2
+CONFIG_CLOCK_STM32_PLL_Q_DIVISOR=7
+CONFIG_CLOCK_STM32_AHB_PRESCALER=1
+CONFIG_CLOCK_STM32_APB1_PRESCALER=4
+CONFIG_CLOCK_STM32_APB2_PRESCALER=2
+
+#enable DTS
```
Comment (galak) [inline boards/arm/stm32f469i_disco/stm32f469i_disco_defconfig]: you can remove the DTS bit, its on for all ARM Socs now

Comment (superna9999) [inline boards/arm/stm32f469i_disco/stm32f469i_disco_defconfig]: Ok

File: dts/arm/stm32f469i_disco.fixup [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * generated data matches the driver definitions.
+ */
+
+
+#define CONFIG_NUM_IRQ_PRIO_BITS	ARM_V7M_NVIC_E000E100_ARM_NUM_IRQ_PRIORITY_BITS
+
+#define CONFIG_UART_STM32_PORT_3_BASE_ADDRESS	ST_STM32_USART_40004800_BASE_ADDRESS
+#define CONFIG_UART_STM32_PORT_3_BAUD_RATE	ST_STM32_USART_40004800_CURRENT_SPEED
+#define CONFIG_UART_STM32_PORT_3_IRQ_PRI	ST_STM32_USART_40004800_IRQ_0_PRIORITY
+#define PORT_1_IRQ				ST_STM32_USART_40004800_IRQ_0
```
Comment (galak) [inline dts/arm/stm32f469i_disco.fixup]: Can you add in:

#define CONFIG_UART_STM32_PORT_3_NAME		ST_STM32_USART_40004800_LABEL

File: dts/arm/stm32f469i_disco.fixup [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+#define CONFIG_UART_STM32_PORT_6_BASE_ADDRESS	ST_STM32_USART_40011400_BASE_ADDRESS
+#define CONFIG_UART_STM32_PORT_6_BAUD_RATE	ST_STM32_USART_40011400_CURRENT_SPEED
+#define CONFIG_UART_STM32_PORT_6_IRQ_PRI	ST_STM32_USART_40011400_IRQ_0_PRIORITY
+#define PORT_6_IRQ				ST_STM32_USART_40011400_IRQ_0
```
Comment (galak) [inline dts/arm/stm32f469i_disco.fixup]: #define CONFIG_UART_STM32_PORT_6_NAME		ST_STM32_USART_40011400_LABEL

**Review by galak — CHANGES_REQUESTED** [review]
some minor DTS related comments to fixup

(as well as the commit message issue)

File: dts/arm/stm32f469i_disco.fixup (hunk 2 above)
Comment (superna9999) [inline dts/arm/stm32f469i_disco.fixup]: Ok

File: dts/arm/stm32f469i_disco.fixup (hunk 3 above)
Comment (superna9999) [inline dts/arm/stm32f469i_disco.fixup]: Ok

File: dts/arm/stm32f469i_disco.fixup (hunk 2 above)
Comment (superna9999) [inline dts/arm/stm32f469i_disco.fixup]: It breaks my build with:
```
  CC      drivers/serial/uart_stm32.o
In file included from /zephyr/include/arch/arm/arch.h:20:0,
                 from /zephyr/include/arch/cpu.h:15,
                 from /zephyr/include/kernel.h:3672,
                 from /zephyr/drivers/serial/uart_stm32.c:19:
/zephyr/samples/hello_world/outdir/stm32f469i_disco/include/generated/generated_dts_board.h:61:0: warning: "CONFIG_UART_STM32_PORT_3_NAME" redefined
 #define CONFIG_UART_STM32_PORT_3_NAME  ST_STM32_USART_40004800_LABEL
 ^
In file included from <command-line>:0:0:
/zephyr/samples/hello_world/outdir/stm32f469i_disco/include/generated/autoconf.h:37:0: note: this is the location of the previous definition
 #define CONFIG_UART_STM32_PORT_3_NAME "UART_3"

```

**Review by dbkinder — APPROVED** [review]
+1 for docs

**Comment by galak** [comment]
Sorry about that PR #211 is going to make changes that will impact things.

**Comment by galak** [comment]
Ok, PR #211 is merged so the changes I suggested should make sense now.

File: dts/arm/stm32f469i_disco.fixup (hunk 2 above)
Comment (superna9999) [inline dts/arm/stm32f469i_disco.fixup]: Ok, done after #211 is merged

File: dts/arm/stm32f469i_disco.fixup (hunk 3 above)
Comment (superna9999) [inline dts/arm/stm32f469i_disco.fixup]: Ok

**Review by galak — APPROVED** [review]

**Comment by Azolo** [comment]
This finding is **incorrect**. I've analyzed the Zephyr fdtable implementation and `zvfs_get_fd_obj()` does **NOT** increment the refcount.

## Evidence from `lib/os/fdtable.c`:

```c
void *zvfs_get_fd_obj(int fd, const struct fd_op_vtable *vtable, int err)
{
    struct fd_entry *entry;
    
    if (_check_fd(fd) < 0) {  // ← Only READS refcount, doesn't modify
        return NULL;
    }
    
    entry = &fdtable[fd];
    return entry->obj;  // ← Just returns pointer, NO z_fd_ref() call!
}
```

The `_check_fd()` helper only **reads** the refcount to verify the FD is valid (lines 132):
```c
if (!atomic_get(&fdtable[fd].refcount)) {  // READ only, not increment
    errno = EBADF;
    return -1;
}
```

## Refcount is only modified by:
- `z_fd_ref()` - increments (called by `zvfs_reserve_fd`)
- `z_fd_unref()` - decrements (called by `zvfs_free_fd`)

## Conclusion:
There is **NO reference leak**. `zvfs_get_fd_obj()` is a read-only query function that doesn't acquire any resources that need releasing.

The FD monitoring code is correct as-is. 👎

**Pull request merged**